lib/tasks/populate_canonical_models.rake

```diff
@@ -89,71 +119,97 @@ namespace :canonical_models do
     end
 
     desc 'Populate or update canonical building and smithing materials from JSON data'
-    task canonical_materials: :environment do
+    task :canonical_materials, [:preserve_existing_records] => :environment do |_t, args|
       Rails.logger.info 'Populating canonical materials...'
 
+      args.with_defaults(preserve_existing_records: false)
+
       canonical_materials = JSON.parse(File.read(Rails.root.join('lib', 'tasks', 'canonical_models', 'canonical_materials.json')), symbolize_names: true)
 
       ActiveRecord::Base.transaction do
+        if FALSEY_VALUES.include?(args[:preserve_existing_records])
+          item_codes = canonical_materials.pluck(:item_code)
+          CanonicalMaterial.where.not(item_code: item_codes).destroy_all
+        end
+
         canonical_materials.each do |canonical_material_attributes|
-          material = CanonicalMaterial.find_or_initialize_by(name: canonical_material_attributes[:name])
+          material = CanonicalMaterial.find_or_initialize_by(item_code: canonical_material_attributes[:item_code])
           material.assign_attributes(canonical_material_attributes)
           material.save!
         rescue ActiveRecord::RecordInvalid => e
-          Rails.logger.error "Validation error saving canonical material \"#{canonical_material_attributes[:name]}\": #{e.message}"
+          Rails.logger.error "Validation error saving canonical material \"#{canonical_material_attributes[:item_code]}\": #{e.message}"
           raise e
         rescue StandardError => e
-          Rails.logger.error "Unknown error saving canonical material \"#{canonical_material_attributes[:name]}\": #{e.message}"
+          Rails.logger.error "Unknown error saving canonical material \"#{canonical_material_attributes[:item_code]}\": #{e.message}"
           raise e
         end
       end
     end
 
     desc 'Populate or update canonical jewelry items from JSON data'
-    task canonical_jewelry: :environment do
+    # rubocop:disable Layout/BlockAlignment
```

Comment:
Ordinarily if a cop causes problems, we disable it in the `.rubocop.yml` file. This time, though, I was conflicted enough about my decision to break the block-starting line into multiple lines that I didn't want to expressly accommodate that kind of code style in our cops. Especially since it's only Rake tasks, I think it makes sense to only disable it here.